src/python/pants/binaries/binary_util.py

```diff
@@ -199,14 +214,9 @@ def _select_binary_stream(self, name, binary_path, fetcher=None):
     :raises: :class:`pants.binary_util.BinaryUtil.BinaryNotFound` if no binary of the given version
       and name could be found for the current platform.
     """
-
-    if not self._baseurls:
-      raise self.NoBaseUrlsError(
-          'No urls are defined for the --pants-support-baseurls option.')
     downloaded_successfully = False
     accumulated_errors = []
-    for baseurl in OrderedSet(self._baseurls):  # De-dup URLS: we only want to try each URL once.
-      url = posixpath.join(baseurl, binary_path)
+    for url in urls:  # De-dup URLS: we only want to try each URL once.
```

Comment:
The comment references the fact that we put the urls into a set. You've removed that for whatever reason (why?) so the comment is no longer relevant.